spec-tests/src/test/java/org/candlepin/spec/entitlements/EntitlementCertificateV3SpecTest.java

```diff
@@ -475,6 +477,59 @@ public void shouldEncodeManyContentUrls() {
             );
     }
 
+    @Test
+    @OnlyInStandalone
+    public void shouldIncludeEntitlementNamespaceForNamespacedProducts() {
+        ProductDTO product = ownerProductApi.createProduct(this.owner.getKey(), Products.randomEng());
+        PoolDTO pool = ownerApi.createPool(this.owner.getKey(), Pools.random(product));
+
+        List<EntitlementDTO> entitlements = consumerApi.bindPoolSync(system.getUuid(), pool.getId(), 1);
+
+        assertThat(entitlements)
+            .isNotNull()
+            .hasSize(1);
+
+        assertThatCert(X509Cert.fromEnt(entitlements.get(0)))
+            .hasEntitlementNamespace(this.owner.getKey());
+    }
+
+    @Test
+    @OnlyInHosted
+    public void shouldNotIncludeEntitlementNamespaceForGlobalProducts() {
+        // Create upstream data
+        ProductDTO product = hostedTestApi.createProduct(Products.randomEng());
+        String productId = product.getId();
+
+        SubscriptionDTO sub = hostedTestApi.createSubscription(Subscriptions.random(this.owner, product));
+
+        // Refresh to pull it in locally
+        AsyncJobStatusDTO refreshJob = ownerApi.refreshPools(this.owner.getKey(), false);
+        assertThatJob(refreshJob)
+            .isNotNull()
+            .terminates(client)
+            .isFinished();
+
+        // Find the pool in the org so we can bind it
+        List<PoolDTO> pools = ownerApi.listOwnerPools(this.owner.getKey());
+        assertThat(pools)
+            .isNotNull()
+            .hasSizeGreaterThanOrEqualTo(1);
+
+        PoolDTO pool = pools.stream()
+            .filter(elem -> productId.equals(elem.getProductId()))
+            .findAny()
+            .get();
+
+        List<EntitlementDTO> entitlements = consumerApi.bindPoolSync(system.getUuid(), pool.getId(), 1);
+
+        assertThat(entitlements)
+            .isNotNull()
+            .hasSize(1);
```

Comment:
We can use `singleElement()` here

Reply:
Yeah, but unless I'm actually mapping the collection to the element for further validation, I tend to prefer just specifying the size.